numpy/__init__.cython-30.pxd

```diff
@@ -806,6 +806,10 @@ cdef extern from "numpy/ndarraytypes.h":
         NPY_DATETIMEUNIT base
         int64_t num
 
+    ctypedef struct npy_datetimestruct:
+        int64_t year
+        int32_t month, day, hour, min, sec, us, ps, as
```

Comment:
Oh, you do need to duplicate the function signatures here of course, but with the magic comment it is now part of this header (I think).

Reply:
Everything duplicated here is inside a `cdef extern from "numpy/foo.h":` block for some "foo".  what im trying to ask is what "foo" is here

Reply:
Foo is this block, I think.  Whatever all of the other functions are like `PyArray_FromArray`, etc.